source/uwp/Renderer/lib/AdaptiveCardRendererComponent.cpp

```diff
@@ -249,6 +255,87 @@ namespace AdaptiveNamespace
         m_defaultResourceDictionary = resourceDictionary;
     }
 
+    std::wstring SubstituteColorsForActionSentimentResourceDictionary(int accentColor, int attentionColor)
+    {
+        std::wstring hexAccentColor;
+        THROW_IF_FAILED(ColorToWString(accentColor, hexAccentColor));
+
+        int clearAccentColor = GenerateClearerColor(accentColor);
+        std::wstring hexClearAccentColor;
+        THROW_IF_FAILED(ColorToWString(clearAccentColor, hexClearAccentColor));
+
+        std::wstring hexAtentionColor;
+        THROW_IF_FAILED(ColorToWString(attentionColor, hexAtentionColor));
+
+        int clearAttentionColor = GenerateClearerColor(attentionColor);
+        std::wstring hexClearAttentionColor;
+        THROW_IF_FAILED(ColorToWString(clearAttentionColor, hexClearAttentionColor));
+
+        // The current string contains %s for denoting the places where the color values will be replaced
+        // The color values are formatted as #XXXXXXXX (9 characters) so we add 7 characters per substituted value
+        size_t originalLength = 0;
+        THROW_IF_FAILED(StringCbLengthW(c_defaultActionSentimentResourceDictionary, STRSAFE_MAX_CCH, &originalLength));
+        
+        int expectedFinalLength = originalLength + 7 * 9;
```

Comment:
nit: this is okay since compiler will do constant folding, but why don't simply write 63 instead 7 * 8.

Reply:
leaving this math undone eases cognitive load on future readers. comment above explains how the math is valid, and the reader doesn't have to multiply in their head to validate. :)

---
In reply to: [226454119](https://github.com/Microsoft/AdaptiveCards/pull/2036#discussion_r226454119) [](ancestors = 226454119)